applications/launchpad_v2/src/components/Tag/Tag.test.tsx

```diff
@@ -0,0 +1,44 @@
+import { render, screen } from '@testing-library/react'
+import { ThemeProvider } from 'styled-components'
+
+import themes from '../../styles/themes'
+import Tag from './'
+import SVGCheck from '../../styles/Icons/Check'
+
+describe('Tag', () => {
+  it('should render Tag component without crashing', () => {
+    render(
+      <ThemeProvider theme={themes.light}>
+        <Tag icon={<SVGCheck />}>Testing</Tag>
+      </ThemeProvider>,
+    )
+
+    const el = screen.getByText('Testing')
+    expect(el).toBeInTheDocument()
+    expect(el).toHaveStyle('fontSize: 12')
+  })
+
+  it('should render the correct tag variant', () => {
+    render(
+      <ThemeProvider theme={themes.light}>
+        <Tag icon={<SVGCheck />} variant='large'>
+          Testing
+        </Tag>
+      </ThemeProvider>,
+    )
+
+    const el = screen.getByText('Testing')
+    expect(el).toHaveStyle('fontSize: 14')
```

Comment:
This test case is tricky to handle, because it would be ideal to avoid checking the implementation details and also do not check the specific style prop like `fontSize` or `color`. But still it's good to test variants somehow.

Because there is already a reference to the `themes.light` in the code, then you can try to find the `fontSize` value in theme, instead of arbitrary assigning `14px`.

Following this and to increase the test coverage, you can add similar test cases for the `type` prop. My suggestion would be:

1. Add `data-testid` to the `<TagContainer data-testid='tag-cmp'`, so then we can easily pick up the element in tests with `screen.getByTestId('tag-cmp')`
2. Add for each type variation a new test like this:

```
it('should render the warning tag with style read from the theme', () => {
  render(
      <ThemeProvider theme={themes.light}>
        <Tag icon={<SVGCheck />} type='warning'>
          Testing
        </Tag>
      </ThemeProvider>,
    )
    
    const warningStyle = themes.light.warning
    
    const el = screen.getByText('Testing')
    expect(el).toHaveStyle(`backgroundColor: ${warningStyle}`)
})
```

Reply:
👍